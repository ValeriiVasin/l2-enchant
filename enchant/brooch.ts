import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const broochEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.TwoToOne,
  items: [Item.Brooch_1, Item.Brooch_2, Item.Brooch_3, Item.Brooch_4, Item.Brooch_5],
  chances: [25, 25, 25, 25],
});
