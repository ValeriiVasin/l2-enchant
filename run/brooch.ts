import { runWithResolution } from '../helpers/run-with-resolution';
import { Item } from '../items';

runWithResolution(
  Item.Brooch_1,
  [Item.Brooch_2, Item.Brooch_3, Item.Brooch_4, Item.Brooch_5],
  1000,
);
