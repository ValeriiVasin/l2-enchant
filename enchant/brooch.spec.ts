import { broochEnchantMap } from './brooch';

it('enchant map: brooch', () => {
  expect(broochEnchantMap).toMatchSnapshot();
});
