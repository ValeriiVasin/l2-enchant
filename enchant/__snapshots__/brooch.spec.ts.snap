// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: brooch 1`] = `
Map {
  "Брошь Ур.1" => Object {
    "cost": "Брошь Ур.1",
    "item": "Брошь Ур.1",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Брошь Ур.2",
      },
      Object {
        "chance": 75,
        "item": "Брошь Ур.1",
      },
    ],
  },
  "Брошь Ур.2" => Object {
    "cost": "Брошь Ур.2",
    "item": "Брошь Ур.2",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Брошь Ур.3",
      },
      Object {
        "chance": 75,
        "item": "Брошь Ур.2",
      },
    ],
  },
  "Брошь Ур.3" => Object {
    "cost": "Брошь Ур.3",
    "item": "Брошь Ур.3",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Брошь Ур.4",
      },
      Object {
        "chance": 75,
        "item": "Брошь Ур.3",
      },
    ],
  },
  "Брошь Ур.4" => Object {
    "cost": "Брошь Ур.4",
    "item": "Брошь Ур.4",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Брошь Ур.5",
      },
      Object {
        "chance": 75,
        "item": "Брошь Ур.4",
      },
    ],
  },
}
`;
